### ISSUE TITLE
Improve stack traces on error in standalone mode

### ISSUE THREAD

**Issue by octachrome** [webdriverio/webdriverio#1657]
## The problem

When an error occurs, it is hard to find out which source file/line number it occurred at. Could something like Bluebird's `Promise.longStackTraces` be implemented to help with this?

## Environment

* WebdriverIO version: 4.2.16
* Node.js version: 6.2.0
* Standalone mode

## Details

Given a simple example:

    var webdriverio = require('webdriverio');
    var options = {
        desiredCapabilities: {
            browserName: 'chrome'
        }
    };
     
    webdriverio
        .remote(options)
        .init()
        .url('http://www.google.com')
        .waitForVisible('#flubber')
        .end()
        .catch(function (err) {
            console.log(err.stack);
        });

It gives the following error:

    Error: element (#flubber) still not visible after 500ms
        at elements("#flubber") - c:\dev\code\wdtest\node_modules\webdriverio\build\lib\commands\isVisible.js:54:17
        at isVisible("#flubber") - c:\dev\code\wdtest\node_modules\webdriverio\build\lib\commands\waitForVisible.js:37:22

There is no reference to which call to waitForVisible caused the problem. If my test calls waitForVisible many times, this makes debugging really hard.

Compare with this:

    var Promise = require('bluebird');
    Promise.longStackTraces();

    Promise.resolve()
        .then(function () {
            // do something
        })
        .then(function () {
            throw new Error('BOOM');
        });

Which gives the following error:

    Unhandled rejection Error: BOOM
        at c:\dev\code\wdtest\test.js:9:15
        at processImmediate [as _immediateCallback] (timers.js:367:17)
    From previous event:
        at Object.<anonymous> (c:\dev\code\wdtest\test.js:8:6)
        at Module._compile (module.js:460:26)
        at Object.Module._extensions..js (module.js:478:10)
        at Module.load (module.js:355:32)
        at Function.Module._load (module.js:310:12)
        at Function.Module.runMain (module.js:501:10)
        at startup (node.js:129:16)
        at node.js:814:3

You not only get a trace of where the error is thrown, but also a trace of where the promise was created that led to the error.

I thought maybe I could make WebdriverIO use Bluebird promises, like this:

    global.Promise = require('bluebird')
    var webdriverio = require('webdriverio')

But Babel does not like Bluebird promises because they fail its compatibility test, and so it chooses to use its own polyfill instead.

**Comment by christian-bromann** [comment]
@octachrome have you tried the wdio testrunner? There you can better stack traces with exact filename and line number of the breaking command. It is quite more difficult in standalone mode when commands are chained like this.

**Comment by octachrome** [comment]
@christian-bromann I haven't tried the testrunner. I'm a bit reluctant to add another layer of abstraction to my tests just to get stack traces, not least because it would take a significant amount of time to port all the test cases across. But I will take a look. Thanks.

**Comment by christian-bromann** [comment]
> I'm a bit reluctant to add another layer of abstraction to my tests just to get stack traces

The wdio testrunner is not another layer of abstraction. It takes a lot of initialisation and boilerplate work from you so that you can concentrate on writing test and not on setting up your testflow. For example it will run your tests in parallel, allows you to include services like [`selenium-standalone`](http://webdriver.io/guide/services/selenium-standalone.html) and makes [tests with writing page objects](https://github.com/webdriverio/webdriverio/blob/master/examples/pageobject/specs/dynamic.spec.js) really easy.

We try to format the stacktraces because you usually get bombed by traces you are not interested in. That works pretty well with the wdio testrunner but apparently not as good as in standalone mode. Bluebird is a great Promise library but we already use `q` so adding another promise library won't help use here. I will try to look into this.

**Comment by christian-bromann** [comment]
Duplicate of #1609
